ToolBox/Polygon/PolygonWebSocketClientWrapper.cs

```diff
@@ -0,0 +1,138 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using Newtonsoft.Json;
+using QuantConnect.Brokerages;
+using QuantConnect.Logging;
+
+namespace QuantConnect.ToolBox.Polygon
+{
+    public class PolygonWebSocketClientWrapper : WebSocketClientWrapper
+    {
+        private const string BaseUrl = "wss://socket.polygon.io";
+
+        private readonly string _apiKey;
+        private readonly ISymbolMapper _symbolMapper;
+        private readonly SecurityType _securityType;
+        private readonly Action<string> _messageHandler;
+
+        public PolygonWebSocketClientWrapper(string apiKey, ISymbolMapper symbolMapper, SecurityType securityType, Action<string> messageHandler)
+        {
+            _apiKey = apiKey;
+            _symbolMapper = symbolMapper;
+            _securityType = securityType;
+            _messageHandler = messageHandler;
+
+            var url = GetWebSocketUrl(securityType);
+            Initialize(url);
+
+            Open += OnOpen;
+            Closed += OnClosed;
+            Message += OnMessage;
+            Error += OnError;
+
+            Connect();
+        }
+
+        public void Subscribe(Symbol symbol, bool subscribe)
```

Comment:
Maybe we should have an `Unsubscribe` public method too + an internal private method with the `bool subscribe`